### ISSUE TITLE
Fix `declaration-property-value-no-unknown` false positives for `anchor()` of `inset`

### ISSUE THREAD

**Issue by otomad** [stylelint/stylelint#8609]
### What minimal example or steps are needed to reproduce the bug?

```css
a {
	top: anchor(top);
	right: anchor(right);
	bottom: anchor(bottom);
	left: anchor(left);
	inset: anchor(center);
	inset-block: anchor(center);
	inset-inline: anchor(center);
	inset-block-start: anchor(start);
	inset-block-end: anchor(end);
	inset-inline-start: anchor(start);
	inset-inline-end: anchor(end);
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
    "rules": {
        "declaration-property-value-no-unknown": true
    }
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLjiADvCkpmABYQBOAFPgQJQDcWuVAlgObk7GkXUaHbjmascAIwh4IAWz5lKtKTNljsOADYwAZrxKLB2vetztMAZxj7+SmmBiYcMKqZzmrOALQTNEMADWCgK0Dk4ubh7WXuaa5jDBdmHOriwaUd6+-gFeFjhIVDaGtHkFomlmltFZgV6OUImC9ZFV3rHxufmFjSVd5eIZMZhxmDB1mA0GITTNaQC+IAA0IDrs2gBySLJwiDAAHlsE2gB0YBYWS+AQmKucCCAYmCjPKOggVACu2hZvxI8vANeIFgYE0BSQ7muXgIVEILhwAE8vAA3JCaD5jTAQLwfTABLEAd2wCBQOE+MCwALmWAWy0gNy4ADFqLIIfcAFYWa6XWAEC6If5AvII7QjHC-IFg5x5N6LSlCxGi8zeem3TqkKAFKASt5SmAykA0kBzIA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "16.20.0"
}
```

### What did you expect to happen?

CSS properties `top`, `right`, `bottom`, `left`, `inset`, `inset-block`, `inset-inline`, `inset-block-start`, `inset-block-end`, `inset-inline-start`, `inset-inline-end` should support `anchor()` function.

[MDN Reference](https://developer.mozilla.org/en-US/docs/Web/CSS/anchor#properties_that_accept_anchor_function_values)

### What actually happened?

The following problems were reported:

```
example.css
2:7-18	error	Unexpected unknown value "anchor(top)" for property "top"	(declaration-property-value-no-unknown)
3:9-22	error	Unexpected unknown value "anchor(right)" for property "right"	(declaration-property-value-no-unknown)
4:10-24	error	Unexpected unknown value "anchor(bottom)" for property "bottom"	(declaration-property-value-no-unknown)
5:8-20	error	Unexpected unknown value "anchor(left)" for property "left"	(declaration-property-value-no-unknown)
6:9-23	error	Unexpected unknown value "anchor(center)" for property "inset"	(declaration-property-value-no-unknown)
7:15-29	error	Unexpected unknown value "anchor(center)" for property "inset-block"	(declaration-property-value-no-unknown)
8:16-30	error	Unexpected unknown value "anchor(center)" for property "inset-inline"	(declaration-property-value-no-unknown)
9:21-34	error	Unexpected unknown value "anchor(start)" for property "inset-block-start"	(declaration-property-value-no-unknown)
10:19-30	error	Unexpected unknown value "anchor(end)" for property "inset-block-end"	(declaration-property-value-no-unknown)
11:22-35	error	Unexpected unknown value "anchor(start)" for property "inset-inline-start"	(declaration-property-value-no-unknown)
12:20-31	error	Unexpected unknown value "anchor(end)" for property "inset-inline-end"	(declaration-property-value-no-unknown)
```

### Do you have a proposal to fix the bug?

Update syntax in CSSTree.

**Comment by jeddy3** [comment]
@otomad Thanks for the report and for using the template.

> Update syntax in CSSTree.

Yes, that's correct. This will be fixed when the syntax is updated in our upstream parser. Interestingly, it seems the [`<anchor>` type is there](https://csstree.github.io/docs/syntax/#Type:anchor()&hltype=Function&hlname=anchor), but it's not used by anything.

In the meantime, you can extend the syntax yourself using the [`languageOptions`](https://stylelint.io/user-guide/configure#languageoptions) configuration property, e.g.:

```json
{
  "languageOptions": {
    "syntax": {
      "properties": { 
        "top": "| <anchor()>"
      }
    }
  },
  "extends": [
    "stylelint-config-standard"
  ]
}
```

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLjiADvCkpmABYQBOAFPgQJQDcWAviADQgBmAlgDYwAckgC2cRDAAeYgoIB0YAM5LO4CJj4BzBCAyYUKdCH6ktAVyRaYAeQI5eGpceL7Dh40oCemHEikuaFju7sYEVIQwVA4wzghoRgYhoSD0gcYAPigAPKQU1DQMAHzGwcmsZYYVSawcZcbSODCYUHHEANqVRiBKOF6C-Ly+ALSQmrxaw72kUEhUUKVJALpsamPaAGLUokg4ugBWShpqsASqiG7dvf0wg77pJruxOMZ1SZ59A0M4oxraU34WnMFvFjKYmr1FuxWEA)

**Comment by otomad** [comment]
For temporary use, you can extend the syntax yourself. For someone who want to use it:

```json
{
	"languageOptions": {
		"syntax": {
			"properties": {
				"top": "| <anchor()>",
				"right": "| <anchor()>",
				"bottom": "| <anchor()>",
				"left": "| <anchor()>",
				"width": "| <anchor-size()>",
				"height": "| <anchor-size()>"
			},
			"types": {
				"self-position": "| anchor-center"
			}
		}
	}
}
```

Note: This patch also add `anchor-center` value to properties like `justify-self`, `align-self`, `place-self`, etc.

**Comment by tobiasbaehr** [comment]
For copy and paster like me, a fixed version.  ;-)

Fixed version, see above.

**Comment by otomad** [comment]
> For copy and paster like me, a fixed version. ;-)

@tobiasbaehr Sorry, I apologize for the mistaken I've made earlier. "self-position" should be under "types" instead of "properties".
